packages/react-components/react-charts-preview/library/src/components/CommonComponents/Popover.types.ts

```diff
@@ -0,0 +1,41 @@
+import { YValueHover } from '../../index';
+
+export interface PopoverComponentProps {
+  clickPosition?: { x: number; y: number };
+  isPopoverOpen?: boolean;
+  xCalloutValue?: string;
+  legend?: string | number | Date;
+  yCalloutValue?: string;
+  YValue?: string | number | Date;
+  XValue?: string;
+  color?: string;
+  culture?: string;
+  customCallout?: {
+    customizedCallout?: JSX.Element;
+    customCalloutProps?: PopoverComponentProps;
+  };
+  isCalloutForStack?: boolean;
+  xAxisCalloutAccessibilityData?: { ariaLabel?: string; data?: string };
+  hoverXValue?: string | number;
+  YValueHover?: YValueHover[];
+  descriptionMessage?: string;
+  ratio?: [number, number];
+  isCartesian?: boolean;
+}
+
+export interface PopoverComponentStyles {
```

Comment:
In v9, styles are provided through the className and the styles hooks. The className is built via makeStyles and mergeClassNames. This avoids having tons of extra types for custom style objects and the styles can be atomicized, deduped, and minified at build time.

Reply:
We are aware about the paradigm and are using the same approach in use**Styles.styles.ts.
But we support customization of styles for individual sub components. These props are to support those scenarios depending on the user needs.
We will rationalize this list over next few iterations.